billing-report-agent/src/main/java/com/epam/pipeline/billingreportagent/service/impl/converter/AbstractStoragePricingService.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017-2020 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.billingreportagent.service.impl.converter;
+
+import com.epam.pipeline.billingreportagent.model.billing.StoragePricing;
+import com.epam.pipeline.entity.region.CloudProvider;
+import lombok.extern.slf4j.Slf4j;
+
+import java.math.BigDecimal;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+@SuppressWarnings("checkstyle:magicNumber")
+public abstract class AbstractStoragePricingService {
+
+    public static final int CENTS_IN_DOLLAR = 100;
+    public static final int BYTES_TO_GB = 1 << 30;
+    public static final int PRECISION = 5;
+
+    private final Map<String, StoragePricing> storagePriceListGb = new HashMap<>();
+    private BigDecimal defaultPriceGb;
+    private final String storageServiceGroup;
+
+    public AbstractStoragePricingService(final String storageServiceGroup) {
+        this.storageServiceGroup = storageServiceGroup;
+        updatePrices();
+    }
+
+    protected abstract void loadFullPriceList() throws Exception;
```

Comment:
I think it is better to actually return something from this method since it is called only form `updatePrices` method where you can assign prices to some inner field.

Reply:
Done, returning `Map` with pricing for regions from now